modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rule-builder/editor/Editor.es.js

```diff
@@ -0,0 +1,833 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {
+	act,
+	cleanup,
+	fireEvent,
+	render,
+	waitForElement,
+} from '@testing-library/react';
+import Color from 'dynamic-data-mapping-form-field-type/ColorPicker/ColorPicker.es';
+import Date from 'dynamic-data-mapping-form-field-type/DatePicker/DatePicker.es';
+import DocumentLibrary from 'dynamic-data-mapping-form-field-type/DocumentLibrary/DocumentLibrary.es';
+import Grid from 'dynamic-data-mapping-form-field-type/Grid/Grid.es';
+import Image from 'dynamic-data-mapping-form-field-type/ImagePicker/ImagePicker.es';
+import Numeric from 'dynamic-data-mapping-form-field-type/Numeric/Numeric.es';
+import RichText from 'dynamic-data-mapping-form-field-type/RichText/RichText.es';
+import Select from 'dynamic-data-mapping-form-field-type/Select/Select.es';
+import Text from 'dynamic-data-mapping-form-field-type/Text/Text.es';
+import React from 'react';
+
+import {Editor} from '../../../../../../src/main/resources/META-INF/resources/data_layout_builder/js/components/rule-builder/editor/Editor.es';
+import {DEFAULT_RULE} from '../../../../../../src/main/resources/META-INF/resources/data_layout_builder/js/components/rule-builder/editor/config.es';
+import {
+	FIELDS,
+	NUMBER_TYPE_FIELDS,
+	OPERATORS_BY_TYPE,
+	STRING_DATATYPE_FIELDS,
+	UPLOAD_TYPE_FIELD,
+} from '../../../../../mock/fields.es';
+
+global.fetch.enableFetchMocks();
+
+const pages = [
+	{
+		label: '1 Page title',
+		name: '0',
+		value: '0',
+	},
+	{
+		label: '2 Page title',
+		name: '1',
+		value: '1',
+	},
+	{
+		label: '3 Page title',
+		name: '2',
+		value: '2',
+	},
+];
+
+const conditions = [
+	{
+		operands: [
+			{
+				type: '',
+				value: '',
+			},
+		],
+		operator: '',
+	},
+	{
+		operands: [
+			{
+				type: '',
+				value: '',
+			},
+		],
+		operator: '',
+	},
+];
+
+const newRule = {
+	...DEFAULT_RULE,
+	conditions,
+};
+
+const defaultProps = (fieldsList = FIELDS) => {
+	return {
+		allowActions: [
+			'auto-fill',
+			'calculate',
+			'enable',
+			'jump-to-page',
+			'require',
+			'show',
+		],
+		dataProvider: [
+			{
+				id: '39421',
+				label: 'Get countries',
+				name: 'Get countries',
+				uuid: 'a6ab90fd-c4ac-b91b-c9a5-cf756170d9d3',
+				value: '39421',
+			},
+		],
+		fields: fieldsList,
+		operatorsByType: OPERATORS_BY_TYPE,
+		pages,
+		rule: DEFAULT_RULE,
+	};
+};
+
+describe('Editor', () => {
+	const originalLiferayLoader = window.Liferay.Loader;
+	afterEach(() => {
+		cleanup();
+	});
+
+	beforeEach(() => {
+		global.fetch.mockResponse(
+			JSON.stringify([
+				{
+					javaScriptModule: 'checkbox_multiple',
+					name: 'checkbox_multiple',
+				},
+				{
+					javaScriptModule: 'color',
+					name: 'color',
+				},
+				{
+					javaScriptModule: 'date',
```

Comment:
I think this will look better in the fields' mock file, such as `FIELDS_TYPES` so that you can use it in the `FormsRuleEditor` test.